mlir/lib/CAPI/RegisterRocMLIR/RegisterRocMLIR.cpp

```diff
@@ -10,11 +10,24 @@
 #include "mlir-c/RegisterRocMLIR.h"
 
 #include "mlir/CAPI/IR.h"
+#include "mlir/Pass/PassManager.h"
 #include "mlir/InitRocMLIRDialects.h"
 #include "mlir/InitRocMLIRPasses.h"
+#include "llvm/Support/CommandLine.h"
 
 void mlirRegisterRocMLIRDialects(MlirDialectRegistry registry) {
   mlir::registerRocMLIRDialects(*unwrap(registry));
 }
 
 void mlirRegisterRocMLIRPasses() { mlir::registerRocMLIRPasses(); }
+
+void mlirRegisterRocMLIROptions() {
+  const char *fakeArgv[] = {"rocMLIR-invoked-as-library",
+                            "--mlir-print-local-scope"};
+  mlir::registerAsmPrinterCLOptions();
+  mlir::registerMLIRContextCLOptions();
+  mlir::registerPassManagerCLOptions();
```

Comment:
Could you extract these `mlir::register*CLOptions()` into a separate function and then also call that in all the various `mlir/tools/rocmlir-*/`s (except perhaps rocmlir-translate)? Might be a useful cleanup while we're here.